### ISSUE TITLE
examples to forecast with multiple feature columns

### ISSUE THREAD

**Issue by s5248** [neocortexdb/functime#30]
as described in title, now there seems to have only example like
```
[entity_col, time_col, price_col]
```
How can I forecast with more columns as:
```
[entity_col, time_col, pct_col, low_col, high_col, open_col, close_col]
```
I had tried data with these column schema but it did not work

**Comment by topher-lo** [comment]
You can find this example in our docs: https://docs.functime.ai/forecasting/#exogenous-regressors.

We use a sklearn-like API: `y` and `X` are separate arguments in `.fit` and `.predict`. This is a pretty common question, so I just added an example into the README as well!

```python
X = (
    y.select([entity_col, time_col])
    .pipe(add_fourier_terms(sp=12, K=6)).collect()
)
X_train, X_future = y.pipe(train_test_split(test_size=3))
forecaster = linear_model(freq="1mo", lags=24)
forecaster.fit(y=y_train, X=X_train)
y_pred = forecaster.predict(fh=3, X=X_future)
```

**Comment by s5248** [comment]
@topher-lo I run your code example, but missing y_train valirbale.
suppose the `y like [entity_col, time_col, pct_col, low_col, high_col, open_col, close_col]` , and target_col is pct_col.
Could you please explain how to get the corresponding y_train? Below is my code:
`
    df = get_sqldf() # custom function
    y=pl.DataFrame(df)
    entity_col, time_col, target_col = y.columns[0: 3]
    X = (
        y.select([entity_col, time_col])
        .pipe(add_fourier_terms(sp=12, K=6)).collect()
    )
    X_train, X_future = y.pipe(train_test_split(test_size=3))
    forecaster = linear_model(freq="1mo", lags=24)
    y_train = y.select([target_col]) # TODO, verify label
    forecaster.fit(y=y_train, X=X_train)
    y_pred = forecaster.predict(fh=3, X=X_future)
`

**Comment by topher-lo** [comment]
Try:

y_train=y.select([entity_col, time_col, pct_change])

**Comment by s5248** [comment]
Thank you, I seem to understand your meaning. The first two columns of `y` must be 'entity' and 'time', and the corresponding label will be automatically selected based on these two columns internally, right? The relevant code snippet is as follows，hope this can help others.

> def make_direct_reduction(
    lags: int, max_horizons: int, y: pl.LazyFrame, X: Optional[pl.LazyFrame] = None
) -> pl.DataFrame:
    idx_cols = y.columns[:2]
    # Defensive lazy
    y = y.lazy()
    X = X.lazy() if X is not None else X
    # Get lags
    y_lag = y.pipe(lag(lags=list(range(1, lags + max_horizons + 1))))
    **X_y = y_lag.join(y, on=idx_cols, how="inner")**.select(
        [*y.columns, *y_lag.columns[2:]]
    )
    # Drop nulls in lagged columns
    if X is not None:
        X_y = _join_X_y(X_y, X)
    # NOTE: Cannot use streaming...
    # Raises error: pyo3_runtime.PanicException: internal error: entered unreachable code
    X_y_final = X_y.collect()
    return X_y_final

**Comment by topher-lo** [comment]
Great! I'll close the issue for now. But other users should be able to find it.
